### ISSUE TITLE
Define root size while creating VM

### ISSUE THREAD

**Issue by swapnilabnave** [Katello/forklift#268]
I have tried this many times, but not able to figure out the option to set disk size other than default (40G).

```ruby
    config.vm.provider :libvirt do |domain|
      domain.memory = 10240
      domain.cpus   = 4
      domain.storage :file, :size => '90G', :type => 'qcow2'
    end
```
This doesn't create a 90GB root volume.
![screenshot from 2016-08-03 21-48-34](https://cloud.githubusercontent.com/assets/1291980/17373035/12c368d4-59c4-11e6-8098-fa0f292f6f8c.png)

I did see a vdb disk with 90+ GB. I am not sure if it's due to the `storage` option.
```ruby
      domain.storage :file, :size => '90G', :type => 'qcow2'
```
![screenshot from 2016-08-03 21-48-25](https://cloud.githubusercontent.com/assets/1291980/17373036/12d2b910-59c4-11e6-9077-ce4ab7ed30e7.png)

--
it doesn't looks that simple to change the VM root disk with libvirt provider.
[libvirt-doc](http://www.rubydoc.info/gems/vagrant-libvirt/0.0.24) didn't help much.
Can someone please help me set the disk size to 90 GB :sweat_smile:

**Comment by ehelms** [comment]
I don't think this is directly possible with libvirt. Libvirt allows you to add a disk that you then have to resize yourself on the VM. To change the disk size of the VM as is, we have to update the backing image (for example the base centos7 box) to have a larger disk size by default.

**Comment by pronix** [comment]
for create custom box with custom size :) https://github.com/vagrant-libvirt/vagrant-libvirt#create-box

**Comment by swapnilabnave** [comment]
Following steps worked for me :raised_hands: 

```sh
##shut down VM/guest
sudo virsh shutdown forklift_centos7-devel
```

```sh
##start virt-manager
sudo virt-manager
```

[GUI: Edit -> Connection Details -> Storage -> default -> <create new qcow2 image with desired size of partition>]

```sh
##Find root partition of old image
➜ sudo virt-filesystems --long --parts --blkdevs -h -a /home/My-VM/forklift_centos7-devel.img    
Name       Type       MBR  Size  Parent
/dev/sda1  partition  83   512M  /dev/sda
/dev/sda2  partition  82   1.0G  /dev/sda
/dev/sda3  partition  83   38G   /dev/sda
/dev/sda   device     -    40G   -
```
In this case partition is `/dev/sda3`

```sh
## Copy content from old `img` to new `qcow2`
sudo virt-resize --expand /dev/sda3 forklift_centos7-devel.img forklift_centos7-devel-extended.qcow2
```

```sh
##Edit guest -> replace the old `img` name with new `qcow2` name
## i.e. replace forklift_centos7-devel.img with forklift_centos7-devel-extended.qcow2
sudo virsh edit forklift_centos7-devel
```

```sh
##verify new qcow2 image is working fine. 
➜ sudo virsh start forklift_centos7-devel
Domain forklift_centos7-devel started
```

Reference
http://libguestfs.org/virt-resize.1.html

**Comment by ehelms** [comment]
@swapnilabnave we had a similar conversation here [1] and I'd be curious how that fits with what you are trying to do and how you are currently solving it.

[1] https://github.com/Katello/forklift/pull/271

**Comment by swapnilabnave** [comment]
@ehelms Umm actually this is not what I thought of. I originally wanted to set disk size more than default 40G.
But the steps I followed were to create a new disk altogether then copy content to it and expand it to whatever size.
I tried `disk_size: 100G` as mentioned by [komidore64](https://github.com/Katello/forklift/pull/271#issue-170442814). But it didn't gave me the space which I expected.
```
[vagrant@centos6-devel ~]$ sudo df -h
Filesystem            Size  Used Avail Use% Mounted on
/dev/mapper/VolGroup00-LogVol00
                      9.0G  1.1G  7.5G  13% /
tmpfs                 4.0G     0  4.0G   0% /dev/shm
/dev/vda1             388M   56M  312M  16% /boot
```

I am not well versed with linux "disk utility" so not sure how to increase the partition. `virt-resize` worked for me.
